packages/flutter/lib/src/material/slider.dart

```diff
@@ -0,0 +1,146 @@
+      : super(key: key);
+
+  final double value;
+  final Color primaryColor;
+  final ValueChanged<double> onChanged;
+
+  _RenderSlider createRenderObject() => new _RenderSlider(
+    value: value,
+    primaryColor: primaryColor,
+    onChanged: onChanged
+  );
+
+  void updateRenderObject(_RenderSlider renderObject, _SliderWrapper oldWidget) {
+    renderObject.value = value;
+    renderObject.primaryColor = primaryColor;
+    renderObject.onChanged = onChanged;
+  }
+}
+
+final Color _kInactiveTrackColor = Colors.grey[400];
+final Color _kActiveTrackColor = Colors.grey[500];
+
+class _RenderSlider extends RenderConstrainedBox {
+  _RenderSlider({
+    double value,
+    Color primaryColor,
+    this.onChanged
+  }) : _value = value,
+       _primaryColor = primaryColor,
+        super(additionalConstraints: const BoxConstraints.tightFor(width: 144.0, height: 2 * _kReactionRadius)) {
+    _drag = new HorizontalDragGestureRecognizer(router: FlutterBinding.instance.pointerRouter)
+      ..onStart = _handleDragStart
+      ..onUpdate = _handleDragUpdate
+      ..onEnd = _handleDragEnd;
+  }
+
+  double get value => _value;
+  double _value;
+  void set value(double newValue) {
+    if (newValue == _value)
+      return;
+    _value = newValue;
+    markNeedsPaint();
+  }
+
+  Color get primaryColor => _primaryColor;
+  Color _primaryColor;
+  void set primaryColor(Color value) {
+    if (value == _primaryColor)
+      return;
+    _primaryColor = value;
+    markNeedsPaint();
+  }
+
+  ValueChanged<double> onChanged;
+
+  HorizontalDragGestureRecognizer _drag;
+  bool _active = false;
+  double _currentDragValue = 0.0;
+
+  void _handleDragStart(Point globalPosition) {
+    _active = true;
+    markNeedsPaint();
+    if (onChanged != null) {
+      _currentDragValue = globalToLocal(globalPosition).x / size.width;
+      onChanged(_currentDragValue.clamp(0.0, 1.0));
+    }
+  }
+
+  void _handleDragUpdate(double delta) {
+    if (onChanged != null) {
+      _currentDragValue += delta / size.width;
+      onChanged(_currentDragValue.clamp(0.0, 1.0));
+    }
+  }
+
+  void _handleDragEnd(Offset velocity) {
+    _active = false;
+    _currentDragValue = 0.0;
+    markNeedsPaint();
+  }
+
+  bool hitTestSelf(Point position) => true;
+
+  void handleEvent(InputEvent event, BoxHitTestEntry entry) {
+    if (event.type == 'pointerdown')
+      _drag.addPointer(event);
+  }
+
+  void paint(PaintingContext context, Offset offset) {
+    final Canvas canvas = context.canvas;
+
+    double trackCenter = offset.dy + size.height / 2.0;
+    double trackLeft = offset.dx;
+    double trackTop = trackCenter - 1.0;
+    double trackBottom = trackCenter + 1.0;
+    double trackRight = offset.dx + size.width;
+    double trackActive = offset.dx + size.width * value;
+
+    Paint primaryPaint = new Paint()..color = _primaryColor;
+    Paint trackPaint = new Paint()..color = _active ? _kActiveTrackColor : _kInactiveTrackColor;
```

Comment:
The colors are in the spec.  The disabled color is the same as the inactive color.